File: content/problems/1557-minimum-number-of-vertices-to-reach-all-nodes/solutions/zero-indegree-vertices/Solution.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n,List<List<Integer>> edges){
        int[] indegree=new int[n];
        for(List<Integer> edge:edges)indegree[edge.get(1)]++;
        List<Integer> answer=new ArrayList<>();
        for(int node=0;node<n;node++)if(indegree[node]==0)answer.add(node);
        return answer;
    }
}

File: content/problems/1557-minimum-number-of-vertices-to-reach-all-nodes/solutions/zero-indegree-vertices/Solution.kt
class Solution {
    fun findSmallestSetOfVertices(n:Int,edges:List<List<Int>>):List<Int>{
        val indegree=IntArray(n);
        for(edge in edges)indegree[edge[1]]++;
        return (0 until n).filter{
            indegree[it]==0
        }
    }
}
